sci-wcalc/air_coil_calc.man

.\" Copyright (c), 2005 Dan McMahill
.\" Do not edit this directly.  Edit the XML source file instead
.\"

.TH air_coil_calc "" "" "Wcalc" "Wcalc Commands"
.B air_coil_calc
- Single layer air core solenoid inductor analysis

.SH SYNOPSIS

[L, Q, SRF, len_out, fill_out, lmax] = 
air_coil_calc(N, len, fill, AWG, rho, dia, freq, flag)


.SH INPUT PARAMETERS

.TP 10
N
: Number of turns
.TP 10
len
: length of the coil (meters)
.TP 10
fill
: ratio of coil length to closewound length
.TP 10
AWG
: wire size (American Wire Gauge)
.TP 10
rho
: resistivity of wire (ohm-m)
.TP 10
dia
: inside diameter of coil (meters)
.TP 10
freq
: analysis frequency (Hertz)
.TP 10
flag
: used to calculate coil length.
 
.RS
.TP 10
flag=0
: the length given by len is used.
.TP 10
flag=1
: the length is calculated from the wire size,
 number of turns and fill input.
.RE


.SH OUTPUT VALUES

.TP 10
L
: calculated inductance (Henries)
.TP 10
Q
: calculated quality factor
.TP 10
SRF
: calculated self resonant frequency (Hertz)
.TP 10
len_out
: length used for the analysis (meters)
.TP 10
fill_out
: fill used for the analysis
.TP 10
lmax
: maximum (closewound) inductance
.SH DESCRIPTION

Function to analyze the electrical characteristics of a single layer,
uniformly spaced, air core, solenoid inductor.  The physical
dimensions of the coil are given as inputs and the inductance, quality
factor, and self resonant frequency are calculated.

.SH EXAMPLE
.nf

.TP
 // number of turns
N=5;
.TP
 // we'll use the fill parameter instead
len=0;
.TP
 // the coil is 20% longer than the minimum
fill=1.2;
.TP
 // wire size
AWG=22;
.TP
 // resistivity of copper (ohm-m)
rho=1.72e-8;
.TP
 // 0.25 inches inside diameter
dia=0.25*0.0254;
.TP
 // 10 MHz operation
f=10e6;
.TP
 // use fill to calculate length
flag=1;
[L,Q,SRF,len_out,fill_out,Lmax] = air_coil_calc(N,len,fill,AWG,rho,dia,f,flag);
disp(sprintf('L=%g nH',L*1e9));
disp(sprintf('Q=%g at %g MHz',Q,f/1e6));
disp(sprintf('Self Resonant Freq=%g MHz',SRF/1e6));
disp(sprintf('Length=%g inches',len_out/0.0254));
disp(sprintf('Fill=%g',fill_out));
disp(sprintf('Lmax=%g nH',Lmax*1e9));
.fi
.SH SEE ALSO
stdio-wcalc(1),
air_coil_calc(), air_coil_syn(), coax_calc(), coax_syn(), coplanar_calc(), coplanar_syn(), coupled_microstrip_calc(), coupled_microstrip_syn(), coupled_stripline_calc(), coupled_stripline_syn(), ic_microstrip_calc(), ic_microstrip_syn(), microstrip_calc(), microstrip_syn(), stripline_calc(), stripline_syn()
.SH AUTHOR

Dan McMahill

.SH BUGS

None known


NEWS
$Id: NEWS,v 1.10 2009/02/28 03:26:18 dan Exp $

Wcalc -- history of user-visible changes. 

Copyright (c) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009 Dan McMahill

Please report bugs on the sourceforge project page.

===============
Version 1.1
===============

Added series/parallel RC equivalent circuit calculator.

Added series/parallel RL equivalent circuit calculator.

Added self and mutual inductance of two rectangular bars.

Added coplanar waveguide model.

Added coupled stripline model.

Corrected the Q calculations for low frequency (not skindepth) region
in air core inductors.  Also corrected Q calculation for conductors
other than copper.

Corrected the calculation of incremental conductance for the
microstrip model.

Corrected a bug in the calculation of conductor losses in the
microstrip model.

Added/enabled code for calculating conductor losses in the coupled
microstrip model.

When compiling with gtk-2.10 or newer, use gtkPrint for printing.
This gives a more professional/standard print dialog and also enables
printing under windows.

Added a preliminary Dutch translation.

Converted the figures to all use solid grayscale fills instead of
pattern fills as the former prints better.

Several updates to the build system to moderize it a bit.  Uses
AM_CONDITIONALS more appropriately, uses AC_PROG_LIBTOOL instead of
AM_PROG_LIBTOOL, etc.

Update the win32 build script to generate an installer that works
correctly under vista.

Add .wc file associations under vista.

Update the octave build infrastructure to work with octave-3.0 and
newer where mex support is built into mkoctfile instead of a
standalone mex shell script.

Switch to GPL licensing.

Create several utility functions to make it faster to build new gtk
forms.  This is only visible to developers.

Improved autogen.sh to check versions of several of the auto* tools to
help make it easier to diagnose any issues which may come up.

Repair compilation with the SunPRO compilers.


===============
Version 1.0
===============
Replaced the alert dialog box in the gtk version with a log
window.  This prevents a problem when many alerts are generated
and the user is presented with many many alert windows.

Added win32 support along with a build script for a windows installer.

Fixed a bug which gave lots of error/warning messages in the cgi
frontend for coupled microstrip.

===============
Version 0.9
===============
First public release.

===============
Version 0.1 
===============
Initial tagged version.  This does good analysis and synthesis
of microstrip.  Considered stable for this purpose.  However,
open/save/print and units other than mils are not supported yet.



octave-wcalc/air_coil_calc.m
%AIR_COIL_CALC    - Single layer air core solenoid inductor analysis 
%  
%  SYNOPSIS
%  
%  [L, Q, SRF, len_out, fill_out, lmax] = air_coil_calc(N, len, fill, AWG, rho,
%  dia, freq, flag)
%  
%  PARAMETERS
%  
%  INPUT PARAMETERS
%  
%    * N : Number of turns
%    * len : length of the coil (meters)
%    * fill : ratio of coil length to closewound length
%    * AWG : wire size (American Wire Gauge)
%    * rho : resistivity of wire (ohm-m)
%    * dia : inside diameter of coil (meters)
%    * freq : analysis frequency (Hertz)
%    * flag : used to calculate coil length.
%        + flag=0 : the length given by len is used.
%        + flag=1 : the length is calculated from the wire size, number of turns
%          and fill input.
%  
%  OUTPUT VALUES
%  
%    * L : calculated inductance (Henries)
%    * Q : calculated quality factor
%    * SRF : calculated self resonant frequency (Hertz)
%    * len_out : length used for the analysis (meters)
%    * fill_out : fill used for the analysis
%    * lmax : maximum (closewound) inductance
%  
%  DESCRIPTION
%  
%  Function to analyze the electrical characteristics of a single layer, uniformly
%  spaced, air core, solenoid inductor. The physical dimensions of the coil are
%  given as inputs and the inductance, quality factor, and self resonant frequency
%  are calculated.
%  
%  EXAMPLE
%  
%   % number of turns
%  N=5;
%   % we'll use the fill parameter instead
%  len=0;
%   % the coil is 20% longer than the minimum
%  fill=1.2;
%   % wire size
%  AWG=22;
%   % resistivity of copper (ohm-m)
%  rho=1.72e-8;
%   % 0.25 inches inside diameter
%  dia=0.25*0.0254;
%   % 10 MHz operation
%  f=10e6;
%   % use fill to calculate length
%  flag=1;
%  [L,Q,SRF,len_out,fill_out,Lmax] = air_coil_calc(N,len,fill,AWG,rho,dia,f,flag);
%  disp(sprintf('L=%g nH',L*1e9));
%  disp(sprintf('Q=%g at %g MHz',Q,f/1e6));
%  disp(sprintf('Self Resonant Freq=%g MHz',SRF/1e6));
%  disp(sprintf('Length=%g inches',len_out/0.0254));
%  disp(sprintf('Fill=%g',fill_out));
%  disp(sprintf('Lmax=%g nH',Lmax*1e9));
%  
%  SEE ALSO
%  
%  air_coil_calc
%  air_coil_syn
%  coax_calc
%  bars_calc
%  coax_syn
%  coplanar_calc
%  coplanar_syn
%  coupled_microstrip_calc
%  coupled_microstrip_syn
%  coupled_stripline_calc
%  coupled_stripline_syn
%  ic_microstrip_calc
%  ic_microstrip_syn
%  microstrip_calc
%  microstrip_syn
%  parallel_rc_p2s
%  parallel_rc_s2p
%  parallel_rl_p2s
%  parallel_rl_s2p
%  stripline_calc
%  stripline_syn
%  
%  AUTHOR
%  
%  Dan McMahill
%  
%  BUGS
%  
%  None known
%  
%  Please report any bugs on the Wcalc Sourceforge Project Page
%  


% Copyright (c) 2001-2009 Dan McMahill
% All rights reserved.
%
% This program is free software; you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation; version 2 of the License.
% 
% This program is distributed in the hope that it will be useful,
% but WITHOUT ANY WARRANTY; without even the implied warranty of
% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
% GNU General Public License for more details.
% 
% You should have received a copy of the GNU General Public License
% along with this program; if not, write to the Free Software
% Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
%
